### PR TITLE
Updates API version for package azure-communication-phonenumber

### PR DESCRIPTION
**Pull request by automagically**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](../CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by lucasrsant — CHANGES_REQUESTED** [review]

File: sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/_generated/operations/_phone_numbers_operations.py [hunk 1]
```diff
@@ -482,56 +804,53 @@ def _update_capabilities_initial(
...
+
+        request = build_update_capabilities_request_initial(
+            phone_number=phone_number,
+            api_version=api_version,
+            content_type=content_type,
+            content=_content,
+            template_url=self._update_capabilities_initial.metadata['url'],
+        )
+        request = _convert_request(request)
+        request.set_json_body(_content)
```
[the remark this answers is not in the document]
Comment (automagically) [reply sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/_generated/operations/_phone_numbers_operations.py:824]: In the autorest version this code was generated there was a bug where it didn't include the Content-length header, this function call was a manual intervention to keep going with the pr while the autorest was being fixed. That said, I've tested the change you proposed and indeed it fixes the issue without calling any 'request' method, furthermore, on the new version of autorest this issue has been fixed, I am  going to re-generate the code in my next commit

File: sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/_generated/aio/operations/_phone_numbers_operations.py [hunk 2]
```diff
@@ -470,56 +489,53 @@ async def _update_capabilities_initial(
...
+
+        request = build_update_capabilities_request_initial(
+            phone_number=phone_number,
+            api_version=api_version,
+            content_type=content_type,
+            content=_content,
+            template_url=self._update_capabilities_initial.metadata['url'],
+        )
+        request = _convert_request(request)
+        request.set_json_body(_content)
```
[the remark this answers is not in the document]
Comment (automagically) [reply sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/_generated/aio/operations/_phone_numbers_operations.py:509]: See other comment for my reply

**Review by lucasrsant — CHANGES_REQUESTED** [review]

File: sdk/communication/azure-communication-phonenumbers/CHANGELOG.md [hunk 3]
```diff
@@ -1,14 +1,8 @@
...
 ## 1.0.2 (Unreleased)
-
-### Features Added
-
-### Breaking Changes
-
-### Bugs Fixed
-
 ### Other Changes
+- Adds support for API verion `2022-01-11-preview2`
```
[the remark this answers is not in the document]
Comment (automagically) [reply sdk/communication/azure-communication-phonenumbers/CHANGELOG.md:5]: Done

[the remark this answers is not in the document]
Comment (lucasrsant) [reply sdk/communication/azure-communication-phonenumbers/CHANGELOG.md:5]: Sorry for the last minute change, but the first phrase should be replaced by: `Support of Denmark (DK) and United Kingdom (GB) phone number acquisitions.`

**Review by lucasrsant — CHANGES_REQUESTED** [review]

**Review by lucasrsant — APPROVED** [review]

File: sdk/communication/azure-communication-phonenumbers/CHANGELOG.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (automagically) [reply sdk/communication/azure-communication-phonenumbers/CHANGELOG.md:9]: Change log updated

**Review by danielortega-msft — APPROVED** [review]

**Review by whisper6284 — APPROVED** [review]

**Pull request merged**